### PR TITLE
Allow POST from staging to test registers

### LINKED ISSUE #82: New register records not showing on reg man tool or register.test API inspector
When I submit and approve my own new register record (as a custodian or advanced user), you see a confirmation alert that the record has been published but you can't see it in the register data view on the Register Management Tool or https://country.test.openregister.org/records

**URL:**

https://managing-registers-staging.cloudapps.digital/country

**Steps to recreate:**

1. Log in as a custodian
2. Click on 'Country' register
3. Click 'Add a new record'
4. Enter details of a new record
5. Click the 'Continue' button
6. Click the checkbox and click 'Submit'
7. Search for the new entry on on the Register Management Tool or https://country.test.openregister.org/records

**Expected result:**

New records visible - in this case 'AB' key and 'Test Country' name

**Actual result:**

screen shot 2017-10-18 at 11 19 35

### PR DESCRIPTION
**Pull request by gidsg**
* Allow POST from staging to test registers
* Refresh data after POST so updates are visible
* Remove unused methods
(Fixes #82)

### CODE REVIEW

File: config/environments/staging.rb [hunk 1]
```diff
@@ -88,6 +88,4 @@
   config.register_phase = :test
   config.register_url = 'test.openregister.org/load-rsf'
   config.register_ssl = true
-  config.register_username = ENV['COUNTRY_REGISTER_USERNAME']
```
Comment (gidsg) [inline config/environments/staging.rb]: these are no longer used now we are reading from the secrets store. We may use them as an override (but on localhost/development only)

File: app/controllers/change_controller.rb [hunk 2]
```diff
@@ -67,31 +67,4 @@ def update
     end
   end
 
-  def post_to_register(register_name, rsf_body)
```
Comment (gidsg) [inline app/controllers/change_controller.rb]: these were refactored in https://github.com/openregister/managing-registers/pull/38 but these were not removed at the time.

Comment (danmitchell-) [inline app/controllers/change_controller.rb]: Good spot removing this 👍

File: config/environments/staging.rb (hunk 1 above)
Comment (danmitchell-) [reply config/environments/staging.rb]: I think we should remove this config (`config.register_username`) from all environment files

Comment (gidsg) [reply config/environments/staging.rb]: My thinking is that they may still be useful on development, e.g. use their existence as an override for local testing, where the credentials should not be in the secrets store.

Comment (gidsg) [reply config/environments/staging.rb]: ...I can delete them in this PR and add them back if I find a use for them, since they are unused currently.

Comment (danmitchell-) [reply config/environments/staging.rb]: I don't think we'll use them. The secrets file is ignored from git so we can have a local copy of that for test/development

**Pull request merged**